src/controllers/confirmRules.controller.js

```diff
@@ -6,11 +6,33 @@ const CookieService = require('../services/cookie.service')
 const LoggingService = require('../services/logging.service')
 const ConfirmRules = require('../models/confirmRules.model')
 
+function timeout (ms) {
```

Comment:
I have managed to reproduce the issue but I'd prefer not to have to use this timeout if possible. Please can you work with the Dynamics guys as they put some tracing on to see if we can get to the bottom of why the completion status doesn't seem to have been saved quickly enough. They should have some time to do this today apparently.

If we end up having to leave this in then a comment should be be added before the timeout() and untilComplete() methods to explain the reason for their existence (in addition to the comment you have already provided lower down at line 62).

Reply:
I have moved this code into the confirmRules model and added comments as requested.